client/network/src/protocol/sync.rs

```diff
@@ -67,6 +67,10 @@ const MAX_IMPORTING_BLOCKS: usize = 2048;
 /// Maximum blocks to download ahead of any gap.
 const MAX_DOWNLOAD_AHEAD: u32 = 2048;
 
+/// Maximum blocks to look backwards. The gap is the difference between the highest block and the
+/// common block of a node.
+const MAX_BLOCKS_TO_LOOK_BACKWARDS: u32 = MAX_DOWNLOAD_AHEAD / 2;
```

Comment:
Does this mean that the node won't be able to recover from a fork longer than 1024 blocks?

Reply:
I added a check that the common block needs to be less than the last finalized block to trigger this behavior.

I can add a test.